### ISSUE TITLE
worker?

### ISSUE THREAD

**Issue by Teressss** [ethereum-mining/ethminer#1021]
Some pools does not allow setting worker as a part of an address (0x4765764676675.worker etc.) and they need just a pure eth address. Claymore has an option -eworker, that is used for cases like this. But in ethminer I've found no option like that. So how to set a worker?

**Comment by AndreaLanfranchi** [comment]
Please post a sample pool.

**Comment by Teressss** [comment]
https://eth.anorak.tech/#/

**Comment by AndreaLanfranchi** [comment]
Apparently you have not read the full documentation of your pool.
anorak.tech states they do allow workername in login info (address.rig_name) as they depict it in the "Stratum mining with ethminer" section.

From their site :

```
Linux
./ethminer -G -SP 1 --opencl-platform 1 -S eth.anorak.tech:8004 -O YOUR_WALLET_ADDRESS.YOUR_RIG_NAME
```

As the -S and -O switches are now deprecated you will only have to change it to :

`./ethminer -G -P stratum1+tcp://<yourethaddress>.<yourworkername>@eth.anorak.tech:8004`

That's it

**Comment by Teressss** [comment]
I posted this pool from my memory, I am mining on 2miners and there is no option for worker in address field. But that is not the point of the problem. Problem is ethminer has no worker option.

**Comment by AndreaLanfranchi** [comment]
It's no issue here.
Worker specification is only part of the login to pool.
There is no need to to have a worker option as each pool states (differently) how to identify a worker.

**Comment by Teressss** [comment]
OK, it is not easy with you, I know. So try to connect to 2miners.com pool with ethminer and set a worker. How do you do that? Impossible.

**Comment by AndreaLanfranchi** [comment]
Impossible uh ?

Launch with :

./ethminer -U -P stratum1+tcp://<ethaddress>.thisisafuckingworkername@eth.2miners.com:2020 --report-hashrate

And voilà (this is my account if you still do not believe).
https://eth.2miners.com/en/account/0xa7e593bde6b5900262cf94e4d75fb040f7ff4727

**Comment by Teressss** [comment]
./ethminer -U -P stratum1+tcp://.thisisafuckingworkername@eth.2miners.com:2020 --report-hashrate

Where is your eth address field??

**Comment by AndreaLanfranchi** [comment]
./ethminer -U -P stratum1+tcp://0xa7e593bde6b5900262cf94e4d75fb040f7ff4727.thisisafuckingworkername@eth.2miners.com:2020 --report-hashrate

**Comment by AndreaLanfranchi** [comment]
Ethereum address followed by a dot and followed by the worker name is (99% of the times) the **standard** way to **login** to stratum pools declaring a worker name.

**Comment by AndreaLanfranchi** [comment]
Can we close this issue please ? (Close button)

**Comment by Teressss** [comment]
will you give me time to test it??

**Comment by Teressss** [comment]
Yes, this way it is working. Thank you. But it's working only because the pool support it in this format. On their pages there is no info on worker settings behind dot and wallet, only info for claymore with -eworker so I thought they do not support it.
But it is still not solving the issue as I mention in my initial post - what to do when pool does not support it and want only pure wallet address and worker set separatelly?

**Comment by Teressss** [comment]
EDIT: when I wrote it is working, I meant I can connect now. But my worker is still not present on pool:

--report-hashrate --api-port 3334 -HWMON --cuda -P stratum1+tcp://0x11eDeF20Ff2C1B19D5c74c88Cddc01BF000000.rig4@clo.2miners.com:3030

**Comment by AndreaLanfranchi** [comment]
>  On their pages there is no info on worker settings behind dot and wallet, only info for claymore with -eworker 


Are you kidding me ?
Do you ever read their FAQ ?
https://eth.2miners.com/en/faq

![2miners-faq](https://user-images.githubusercontent.com/15280851/39116314-6ebf9982-46e4-11e8-8de1-0c436e084652.jpg)

Here is where they depict the syntax address-dot-rigname

**Comment by Teressss** [comment]
https://clo.2miners.com/en/help

**Comment by AndreaLanfranchi** [comment]
> EDIT: when I wrote it is working, I meant I can connect now. But my worker is still not present on pool:

Does you worker have found at least one share ?
As the address you provided is still not present

**Comment by Teressss** [comment]
Of course it did found shares, I am not an idiot even you think it, actually 11 shares in 7 minutes, pool still see no worker. In address I posted I wrote few zeros at the end, I don't want anybody see my real address..

**Comment by Teressss** [comment]
it could be also problem on the pool..

**Comment by Teressss** [comment]
Problem was also that pool does not accept workers with special characters like "-" and in that case all of the workers was displayed under one "0" worker.

**Comment by AndreaLanfranchi** [comment]
It works.
I'm mining with a random address picked from Callisto block explorer.
And here is the result.

https://clo.2miners.com/en/account/0x0073cf1b9230cf3ee8cab1971b8dbef21ea7b595

You see 'worker' ?

I tried first with "thisisafuckingworkername" but I could not see it.
Then I read the help and see they want a string at most of 8 chars.
So i tried with "Worker-1" but still no luck (maybe they do not want uppercases o dashes ???).
Eventually I inserted only worker and voilà.

So this is absolutely not an issue about ethminer.
Ask the pool support about how to handle it.
